src/codeu/chat/client/commandline/Chat.java

```diff
@@ -276,14 +308,142 @@ public void invoke(Scanner args) {
 
       // Find the first conversation with the given name and return its context.
       // If no conversation has the given name, this will return null.
-      private ConversationContext find(String title) {
-        for (final ConversationContext conversation : user.conversations()) {
-          if (title.equals(conversation.conversation.title)) {
-            return conversation;
+    });
+
+    // C-FOLLOW (follows an interest)
+    //
+    // Adds a command that will allow the user to follow users and conversations
+    // in order to get updates
+    //
+    panel.register("i-add", new Panel.Command(){
+      @Override
+      public void invoke(Scanner args){
+        Type type = Type.USER;
+        ArrayList<String> arguments = new ArrayList<>();
+  
+        while(args.hasNextLine()){
+          arguments.add(args.next().toLowerCase().trim());
+        }
+
+        if(arguments.size() == 2){ 
+          
+          // First arguments etermines if an user or conversation will be added to the interest
+          // system
+          // Seconds argument determines if the username or conversation name exists
+          final UserContext userInterest = findUser(arguments.get(1), context); 
+          final Uuid userId = user.user.id;         
+          final ConversationContext convoInterest = find(arguments.get(1), user);
+
+          if(arguments.get(0).equals("u")){
+            type = Type.USER;
+            if(userInterest != null){
+              user.getController().newInterest(userId, userInterest.user.id, type);
+            }else{
+              System.out.format("ERROR: '%s' does not exist", userInterest);
+            }
+          }else if(arguments.get(0).equals("c")){
+            if(convoInterest != null){
+              type = Type.CONVERSATION;
+              user.getController().newInterest(userId,
+                  convoInterest.conversation.id, type);  
+            }else{
+              System.out.format("ERROR: '%s' does not exist", convoInterest);
+            }
+
+          }else{
+            System.out.println("ERROR: Wrong format");
+            return;
+          }
+                  
+        }else{
+          System.out.println("ERROR: Wrong format");
+        }
+      } 
+    });
+
+    // C-UNFOLLOW (unfollows an interest)
+    // 
+    // Adds a command that will allow the user to unfollow users and
+    // conversations to stop getting updates 
+    //
+    panel.register("i-remove", new Panel.Command(){
+      @Override
+      public void invoke(Scanner args){
+
+        ArrayList<String> arguments = new ArrayList<String>();
```

Comment:
Change to`List<String> arguments = ...`. Use interface classes whenever possible.